pennylane/math/sparse_matrix.py

```diff
@@ -0,0 +1,243 @@
+        """Comparison to another ``SparseMatrix`` object.
+        This method compares the data dictionaries.
+
+        Args:
+            other (SparseMatrix): sparse matrix to compare to
+
+        Returns:
+            bool
+        """
+        if not isinstance(other, type(self)):
+            return False
+        return self.data == other.data and self.shape == other.shape
+
+    def __repr__(self):
+        """Representation of this object.
+
+        Return:
+            str
+        """
+        return f"<SparseMatrix: entries={self.nnz}, shape={self.shape}>"
+
+    def __add__(self, other):
+        """Addition of another ``SparseMatrix`` object.
+
+        Args:
+            other (SparseMatrix): sparse matrix to add
+
+        Returns:
+            .SparseMatrix: sum of sparse matrices
+        """
+        if not isinstance(other, type(self)):
+            raise ValueError(f"Cannot add SparseMatrix and {type(other)}.")
+        if self.shape != other.shape:
+            raise ValueError(
+                f"Cannot add SparseMatrix object of different shape, got {other.shape}"
+            )
+
+        new = deepcopy(self)
+        for idx, entry in other.data.items():
+            if idx in new.data:
+                new.data[idx] += entry
+                # if the sum cancelled the element, remove it
+                if qml.math.count_nonzero(new.data[idx]) == 0:
+                    del new.data[idx]
+            else:
+                new.data[idx] = entry
+        return new
+
+    def __sub__(self, other):
+        """Subtraction of another SparseMatrix object.
+
+        Args:
+            other (SparseMatrix): sparse matrix to subtract
+
+        Returns:
+            .SparseMatrix: difference of sparse matrices
+        """
+        return self + -1.0 * other
+
+    def __mul__(self, other):
+
+        if qml.math.count_nonzero(other) == 0:
+            # return an empty matrix
+            return SparseMatrix(self.shape)
+
+        new = deepcopy(self)
+        for idx in new.data.keys():
+            new.data[idx] *= other
+
+        return new
+
+    __rmul__ = __mul__
+
+    def kron(self, other):
+        """An implementation of a sparse kronecker product inspired by ``scipy.sparse.sparse_kron``.
+
+        Args:
+            other (SparseMatrix): another sparse matrix
+
+        Returns:
+            .SparseMatrix: new sparse matrix object representing the kronecker product
+        """
+        if not isinstance(other, SparseMatrix):
+            raise ValueError(
+                f"Can only compute the kronecker product with another SparseMatrix; got {type(other)}"
+            )
+
+        output_shape = (self.shape[0] * other.shape[0], self.shape[1] * other.shape[1])
+
+        # note: we can use vanilla np here because we are only bookkeeping indices
+        row = np.repeat(self.row, other.nnz)
+        col = np.repeat(self.col, other.nnz)
+        entries = np.repeat(list(self.data.values()), other.nnz)
```

Comment:
something else I just thought of - it might make sense to store `values` and `indices` as separate lists, since calling `list(dict.values())` will (I believe?) result in an implicit for loop over the dictionary values.

Although, I suppose this will break, since we are forced to use `data` due to PL assumptions?

Reply:
I will add it to the list of todos!